### PR TITLE
Peterl connected component

### PR DESCRIPTION
**Pull request by peterl-stripe**
@dandekar Here is a PR for connected component of a given node.  Could you review this?

### CODE REVIEW

File: src/main/scala/co/teapot/tempest/typedgraph/SimpleTypedGraph.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import co.teapot.tempest.Node
+import co.teapot.tempest.graph.DirectedGraph
+
+/** A SimpleTypedGraph has a single source node type and single target node type. */
+case class SimpleTypedGraph(sourceNodeType: String, targetNodeType: String, graph: DirectedGraph) extends TypedGraph {
+  override def outNeighbors(node: Node): Traversable[Node] = new Traversable[Node] {
+    def foreach[A](f: Node => A): Unit = {
+      // In use cases like the union graph, we might give this simple graph a node whose type doesn't match, in which
+      // case outNeighbors should do nothing.
```
Comment (ppd-stripe) [inline src/main/scala/co/teapot/tempest/typedgraph/SimpleTypedGraph.scala:11]: What will this return if the node type isn't a valid node type for the graph? Should it return an empty list?

**Comment by ppd-stripe** [comment]
Other than my minor comment/question, looks good to me.

**Comment by ppd-stripe** [comment]
Just to confirm, this PR doesn't have the method for computing all connected components in a graph, right? I didn't see; just want to make sure I didn't miss it.

File: src/main/scala/co/teapot/tempest/typedgraph/SimpleTypedGraph.scala (hunk 1 above)
Comment (peterl-stripe) [reply src/main/scala/co/teapot/tempest/typedgraph/SimpleTypedGraph.scala:11]: Right, the empty list (or technically the empty Traversable, which does nothing when you call foreach on it).  I think that's the right thing to do.

**Comment by peterl-stripe** [comment]
Right, no method yet for computing all connected components in a graph.  I realized that Ashish's disjoint set data structure works on Ints rather than Nodes, so I need to decide if I want to re-implement disjoin set for Nodes, or just repeat this single-component call until all nodes have been covered by one of the components.  I'll do that in a later PR.

**Pull request merged**